src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/client/ClientAdapter.kt

```diff
@@ -76,12 +96,14 @@ abstract class ClientAdapter<C : KubernetesClient>(private val fabric8Client: C)
                     setSslContext(httpClientBuilder, config, trustManager)
                 }
                 .build()
-            return if (ClusterHelper.isOpenShift(kubeClient)) {
-                val osClient = kubeClient.adapt(NamespacedOpenShiftClient::class.java)
-                OSClientAdapter(osClient, kubeClient)
-            } else {
-                KubeClientAdapter(kubeClient)
-            }
+            /**
+             *  Always create kubernetes client.
+             *  Upgrade existing client to OpenShift only async bcs checking if cluster is OpenShift is costly
+             *  and may timeout if cluster is not reachable.
+             *  @see [issue 865](https://github.com/redhat-developer/intellij-kubernetes/issues/865)
+             *  @see ClientAdapter.toOpenShift
+             **/
+            return KubeClientAdapter(kubeClient)
```

Comment:
this is the core of the fix: dont check if the cluster is OpenShift and create a Kubernetes- or OpenShift context. This check is costly and may timeout if the cluster is not reachable.
Create a Kubernetes context that async tries to upgrade to an OpenShift context instead.